draft-ietf-lisp-pubsub.xml

```diff
@@ -294,7 +294,7 @@ Rec +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
           <t>If the Map-Server receives a subscription request for an EID-Prefix not present in the mapping database, it SHOULD follow the same logic described in Section 8.4 of <xref target="RFC9301"></xref> and create a temporary subscription state for the xTR-ID to the least-specific prefix that both matches the original query and does not match any EID-Prefix known to exist in the LISP-capable infrastructure. Alternatively, the Map-Server can instead determine that such subscription request fails, and send a Negative Map-Reply following Section 8.3 of <xref target="RFC9301"></xref>. In both cases, the TTL of the temporary subscription state or the Negative Map-Reply SHOULD be configurable, with a value of 15-minutes being RECOMMENDED. </t>
 
-          <t>The subscription state can also be created explicitly by configuration at the Map-Server (possible when a pre-shared security association exists, see <xref target="security"></xref>). In this case, the initial nonce associated with the xTR-ID (and EID-Prefix) MUST be randomly generated by the Map-Server.</t>
+          <t>The subscription state can also be created explicitly by configuration at the Map-Server (possible when a pre-shared security association exists, see <xref target="security"></xref>). In this case, if the Map-Server has no nonce state to use (e.g. when a single nonce per xTR-ID is not being used, or when this has not been initialized yet), the initial nonce associated with the xTR-ID (and EID-Prefix, if appropriate) MUST be randomly generated by the Map-Server. It is RECOMMENDED to have a configuration option to enable/disable the xTR to accept publication information for EID-Prefixes it did not explicitly subscribed to.</t>
```

Comment:
Let's assume this case: 

> xTR===>Map-Request(EID1) ==> MS
>                       ....
> xTR<====N1:Map-Notify(EID1)===MS
>                     ...
> xTR<====Ni: Map-Notify(EID1)===MS
> xTR===>Map-Request(delete EID1 state) ==> MS
> xTR<====N1:Map-Notify(EID1)=== Attacker

If the xTR is configured to accept notifications for non-explicitly subscribed state, how it will distinguish the last N1 in the example above vs. a legitimate message from the Map-Server?

Reply:
The way the draft reads today, the xTR MUST keep nonce state if it ever has gotten it from a Map-Notify. The draft doesn't say that the local nonce state goes away when the xTR requests the deletion of subscription.

> The xTR MUST keep track of the last nonce seen in a Map-Notify received as a publication from the Map-Server for the EID-Prefix. 

The replayed Map-Notify will have N1 < Ni

P.S. The considerations on persistent storage are carried over from RFC9301

Reply:
> The way the draft reads today, the xTR MUST keep nonce state if it ever has gotten it from a Map-Notify. The draft doesn't say that the local nonce state goes away when the xTR requests the deletion of subscription.
 
Hmm. This is for an active subscription. Not for any random the xTR has subscribed and withdrawn in the past.

Reply:
The only way I see explicit subscriptions would work without side effects is to coordinate out-of-band the creation of the sate/nonce in both the xTR/Map-Server (using for example, NETCONF). Then, things notifications handling at the xTR would follow the normal procedure. 

However, why that out-band provisioning is better than triggering the subscription from the xTR itself is questionable.

Reply:
Deal! 

Please consider adding a statement about if expect any "special" processing at the Map-Server for such state: indicate if by default we disable or enable explicit-subscriptions to be deleted/updated by an xTR. Thanks.

Reply:
Let me know what you think of the proposed changes and feel free to close and merge if they look good to you.

Reply:
> Let me know what you think of the proposed changes and feel free to close and merge if they look good to you.

Yes, I think this one can be closed. Thanks.